Lib/test/test_syntax.py

```diff
@@ -698,17 +698,35 @@ def test_break_outside_loop(self):
 
     def test_yield_outside_function(self):
         self._check_error("if 0: yield",               "outside function")
-        self._check_error("class C:\n    if 0: yield", "outside function")
+        self._check_error("if 1: pass\nelse: yield",   "outside function")
```

Comment:
Could you add a test for the pattern in https://bugs.python.org/issue37289 ir you prefer to do it in another PR?

Reply:
I think that test is unrelated.